src/util/detect-bugs/unregistered-transition.js

```diff
@@ -1,18 +1,21 @@
 // Detect https://issues.chromium.org/issues/360159391
-let dummy = document.createElement("div");
-document.body.appendChild(dummy);
-let property = "--foo-" + Date.now();
-dummy.style.cssText = `${property}: 1; transition: ${property} 1ms step-start allow-discrete`;
 
 /**
  * Detect if the browser is affected by the unregistered transition bug.
  * @see https://issues.chromium.org/issues/360159391
- * @type {Promise<boolean>}
+ * @returns {Promise<boolean>}
  */
-export default new Promise(resolve => {
-	requestAnimationFrame(() => {
-		setTimeout(_ => resolve(true), 30);
-		dummy.addEventListener("transitionstart", _ => resolve(false));
-		dummy.style.setProperty(property, "2");
-	});
-}).finally(_ => dummy.remove());
+export default function detectUnregisteredTransitionBug() {
+	let dummy = document.createElement("div");
+	document.body.appendChild(dummy);
+	let property = "--foo-" + Date.now();
+	dummy.style.cssText = `${property}: 1; transition: ${property} 1ms step-start allow-discrete`;
+
```

Comment:
If I'm reading this right, this will run the test every time the function is called. The idea is the test would still only be run once, the function would cache the result.

Reply:
That's correct, but they are called only once and the result is caches in `detect-bugs.js`. I feel like that's more "correct"?

Reply:
@LeaVerou – I changed the implementation to make `observe` synchronous again in https://github.com/LeaVerou/style-observer/pull/111/commits/25d479c0ac47440c297aded338637a47d64062aa, and ended up doing what you suggested here. :)